packages/server-wallet/src/protocols/close-channel.ts

```diff
@@ -23,13 +23,14 @@ export type ProtocolState = {
   ledgerChannelId?: Bytes32;
 };
 
-const stageGuard = (guardStage: Stage) => (s: State | undefined): s is State =>
-  !!s && stage(s) === guardStage;
-
-const isFinal = stageGuard('Final');
-
 function everyoneSignedFinalState(ps: ProtocolState): boolean {
-  return (ps.app.support || []).every(isFinal) && isFinal(ps.app.latestSignedByMe);
+  return (
+    !!ps.app.latestSignedByMe &&
+    ps.app.latestSignedByMe.isFinal &&
```

Comment:
I don't see why this is needed.
```suggestion
```
- If `ps.app.support.every(isFinal)`, then I have signed a final state
- My wallet should not be signing any other states once it signs a final state.

Reply:
I think it's just because the unit tests provide `undefined` for `support` that I did this

Reply:
The following line should return `false` in that case?